src/API.jl

```diff
@@ -916,6 +916,10 @@ end
 precompile(; kwargs...) = precompile(Context(); kwargs...)
 function precompile(ctx::Context; internal_call::Bool=false, kwargs...)
     Context!(ctx; kwargs...)
+    if !Operations.is_instantiated(ctx)
+        Pkg.instantiate(ctx; allow_autoprecomp=false, kwargs...)
```

Comment:
Can't `instantiate` be called directly? If the env is instantiated it should be a no-op?

Reply:
Yeah, you're right
```
julia> ctx = Pkg.Types.Context();

julia> @time Pkg.Operations.is_instantiated(ctx)
  0.156026 seconds (915.04 k allocations: 69.247 MiB, 8.86% gc time)
true

julia> @time Pkg.Operations.is_instantiated(ctx)
  0.154110 seconds (915.04 k allocations: 69.252 MiB, 7.59% gc time)
true

julia> @time Pkg.instantiate(ctx, allow_autoprecomp = false)
  0.147273 seconds (917.09 k allocations: 69.444 MiB, 3.33% gc time)

julia> @time Pkg.instantiate(ctx, allow_autoprecomp = false)
  0.152403 seconds (917.10 k allocations: 69.454 MiB, 6.80% gc time)
```